apps/web/src/common/modules/widgets/_components/WidgetFormDataTableCardTransformPivotForm.vue

```diff
@@ -0,0 +1,261 @@
+const widgetGenerateState = widgetGenerateStore.state;
+
+const storeState = reactive({
+    dataTables: computed(() => widgetGenerateState.dataTables),
+});
+const state = reactive({
+    isInitiated: true,
+    proxyDataTableInfo: useProxyValue<TransformDataTableInfo>('dataTableInfo', props, emit),
+    proxyFormData: useProxyValue<Omit<PivotOptions, 'dataTableId'>>('formData', props, emit),
+    selectedDataTable: computed(() => storeState.dataTables.find((dataTable) => dataTable.data_table_id === state.proxyDataTableInfo.dataTableId)),
+    labelFieldItems: computed<MenuItem[]>(() => {
+        if (!state.selectedDataTable) return [];
+        const labelsInfo = state.selectedDataTable.labels_info;
+        return Object.keys(labelsInfo).map((key) => ({
+            name: key,
+            label: key,
+        }));
+    }),
+    dataFieldItems: computed<MenuItem[]>(() => {
+        if (!state.selectedDataTable) return [];
+        const dataInfo = state.selectedDataTable.data_info;
+        return Object.keys(dataInfo).map((key) => ({
+            name: key,
+            label: key,
+        }));
+    }),
+    columnFieldInvalid: computed<boolean>(() => {
+        if (state.labelFieldItems.length === 1) return true;
+        return false;
+    }),
+    valueTypeItems: computed<MenuItem[]>(() => [
+        {
+            name: 'auto',
+            label: i18n.t('DASHBOARDS.WIDGET.OVERLAY.STEP_2.AUTO'),
+        },
+        {
+            name: 'fixed',
+            label: i18n.t('DASHBOARDS.WIDGET.OVERLAY.STEP_2.FIXED'),
+        },
+    ]),
+    selectedValueType: 'auto',
+    selectedDynamicFieldMenuItems: computed(() => []),
+    dynamicFieldLoading: false,
+});
+/* Events */
+const handleUpdateCriteria = (value: string) => {
+    state.proxyFormData = {
+        ...state.proxyFormData,
+        fields: {
+            ...state.proxyFormData.fields,
+            data: value,
+        },
+    };
+};
+const handleUpdateColumn = (value: string) => {
+    state.proxyFormData = {
+        ...state.proxyFormData,
+        fields: {
+            ...state.proxyFormData.fields,
+            column: value,
+        },
+    };
+};
+
+const handleChangeValueType = (value: string) => {
+    state.selectedValueType = value;
+    if (value === 'auto') {
+        state.proxyFormData = {
+            ...state.proxyFormData,
+            select: undefined,
+            limit: DEFAULT_TRANSFORM_DATA_TABLE_VALUE_MAP.PIVOT.limit,
+        };
+    } else {
+        state.proxyFormData = {
+            ...state.proxyFormData,
+            select: [],
+            limit: undefined,
+        };
+    }
+};
+
+const handleSelectDynamicFields = (value: MenuItem[]) => {
+    state.proxyFormData = {
+        ...state.proxyFormData,
+        select: value.map((item) => item.name),
+    };
+};
+const handleUpdateLimit = (value: string) => {
+    state.proxyFormData = {
+        ...state.proxyFormData,
+        limit: value,
+    };
+};
```

Comment:
nitpick: updating the merged object(formData) everytime when each property has changed is.. I think, a little dangerous 🤔.
it's hard to track which handler changes that object. That means, it's hard to maintain.
But it's just a nitpick comment and another solution also can have other problems... 😂